sycl/doc/extensions/supported/sycl_ext_intel_esimd/sycl_ext_intel_esimd_functions.md

```diff
@@ -631,6 +638,79 @@ The `byte_offsets` is a vector of any integral type elements, limited in [statef
+```C++
+namespace sycl::ext::intel::esimd {
+enum fence_mask : uint8_t {
+  /// “Commit enable” - wait for fence to complete before continuing.
+  global_coherent_fence = 0x1,
+  /// Flush the instruction cache.
+  l2_flush_instructions = 0x2,
+  /// Flush sampler (texture) cache.
+  l2_flush_texture_data = 0x4,
+  /// Flush constant cache.
+  l2_flush_constant_data = 0x8,
+  /// Flush constant cache.
+  l2_flush_rw_data = 0x10,
+  /// Issue SLM memory barrier only. If not set, the memory barrier is global.
+  local_barrier = 0x20,
+  /// Flush L1 read - only data cache.
+  l1_flush_ro_data = 0x40
+};
+/*fence-1*/template <uint8_t ctrl_mask> void fence();
+
+
+/// The target memory kind for fence() operation.
+enum class memory_kind : uint8_t {
+  global = 0, /// untyped global memory
+  image = 2, /// image (also known as typed global memory)
+  local = 3, /// shared local memory
+};
+/// The cache flush operation to apply to caches after fence() is complete.
+enum class fence_flush_op : uint8_t {
+  none = 0,       /// no operation;
+  evict = 1,      /// R/W: evict dirty lines; R/W and RO: invalidate clean lines
+  invalidate = 2, /// R/W and RO: invalidate all clean lines;
+  clean = 4 /// R/W: dirty lines are written to memory, but retained in
+            /// cache in clean state; RO: no effect.
+};
+/// The scope that fence() operation should apply to.
+enum class fence_scope : uint8_t {
+  /// Wait until all previous memory transactions from this thread are observed
+  /// within the local thread-group.
+  group = 0,
+  /// Wait until all previous memory transactions from this thread are observed
+  /// within the local sub-slice.
+  local = 1,
+  /// Wait until all previous memory transactions from this thread are observed
+  /// in the local tile.
+  tile = 2,
+  /// Wait until all previous memory transactions from this thread are observed
+  /// in the local GPU.
+  gpu = 3,
+  /// Wait until all previous memory transactions from this thread are observed
+  /// across all GPUs in the system.
+  gpus = 4,
+  /// Global memory data-port only: wait until all previous memory transactions
+  /// from this thread are observed at the "system" level.
+  system = 5,
+  /// Global memory data-port only: for GPUs that do not follow
+  /// PCIe Write ordering for downstream writes targeting device memory,
+  /// this op will commit to device memory all downstream and peer writes that
+  /// have reached the device.
+  system_acquire = 6
+};
+
+/*fence-2*/template <memory_kind Kind = memory_kind::global,
+                     fence_flush_op FenceOp = fence_flush_op::none,
+                     fence_scope Scope = fence_scope::group> void fence();
+} // end namespace sycl::ext::intel::esimd
+```
+### Description
+`(fence-1)`: Sets the memory read/write order. This function has pretty limited functionality comparing to `(fence-2)`. It accepts 8-bit `ctrl_mask` containing one or more `fence_mask` enum values in it. It can be used for any Intel GPU.
```

Comment:
```suggestion
`(fence-1)`: Sets the memory read/write order. This function has pretty limited functionality compared to `(fence-2)`. It accepts an 8-bit `ctrl_mask` containing one or more `fence_mask` enum values in it. It can be used for any Intel GPU.
```